iceoryx2-bb/trait-tests/tests/semantic_string_tests.rs

```diff
@@ -154,6 +155,15 @@ mod semantic_string {
         assert_that!(sut.as_bytes(), eq b"a012");
     }
 
+    #[test]
+    fn invalid_utf8_characters_fail<const CAPACITY: usize, Sut: SemanticString<CAPACITY>>() {
```

Comment:
Could you also add tests for `insert`, `remove`, `pop`, etc. for an initial string of "🧐"? It is three bytes wide and should ensure that those functions work properly with utf8

Reply:
No I cannot here, since this is not always a valid semantic string. But it is allowed for path, filename etc - I will add it there.

Reply:
But aren't these tests for the generic `SemanticString`?